lib/gateway-kernel.coffee

```diff
@@ -0,0 +1,94 @@
+child_process = require 'child_process'
+path = require 'path'
+
+_ = require 'lodash'
+uuid = require 'uuid'
+services = require('./jupyter-js-services-shim')
+
+Kernel = require './kernel'
+
+class GatewayKernel extends Kernel
+    constructor: (kernelSpec, session) ->
+        super kernelSpec
+
+        @session = session
+        @session.statusChanged.connect(@_onStatusChange, this)
```

Comment:
I'll double-check, but I remember looking into this when writing the code.

Apparently the second argument (the binding for this) is required by the signalling API being used. If `this` is not passed I'm pretty sure it will cause an error to be thrown.